R/arg.R

```diff
@@ -35,9 +35,10 @@ arg_match <- function(arg, values = NULL) {
     fn <- caller_fn()
     values <- fn_fmls(fn)[[arg_nm]]
     values <- eval_bare(values, get_env(fn))
-  }
-  if (!is_character(values)) {
-    abort("Internal error: `values` must be a character vector.")
+  } else {
+    signal_soft_deprecated(
+      "The `values` argument to `arg_match()` is deprecated as of rlang 0.4.7. Use `arg_match0()` instead."
```

Comment:
I feel slightly uncomfortable about deprecating `values`, but I'll defer to @lionel-

Reply:
It's rarely used, only soft-deprecated, and the deprecation cycles in rlang are very long. For these reasons I think it's ok and this way we encourage using the non-NSE function when performance is important.